tools/MRIProtocolRerunner.pl

```diff
@@ -0,0 +1,206 @@
+#! /usr/bin/perl
+
+=pod
+
+=head1 NAME
+
+MRIProtocolRerunner.pl -- Reruns MRI protocol checks on all existing MINC insertions & violations
+
+=head1 SYNOPSIS
+
+Work in progress...
+
+=cut
+
+
+use strict;
+use warnings;
+use Getopt::Tabular;
+use FindBin;
+use File::Temp qw/ tempdir /;
+
+
+use lib "$FindBin::Bin";
+use NeuroDB::File;
+use NeuroDB::DBI;
+use NeuroDB::MRIProcessingUtility;
+use POSIX;
+use File::Basename;
+
+my $profile = undef;
+my $verbose = 0;
+
+my @opt_table           = (
+    [
+        "-profile", "string", 1, \$profile,
+        "name of config file in ../dicom-archive/.loris_mri"
+    ],
+    ["-verbose", "boolean", 1,   \$verbose, "Be verbose."],
+);
+
+
+my $Help = <<HELP;
+******************************************************************************
+Dicom Validator
+******************************************************************************
+
+Author  :
+Date    :
+Version :
+
+The program does the following
+
+- Gets the location of the uploaded file (.zip,.tar.gz or .tgz)
+
+- Unzips the uploaded file
+
+- Uses the ImagingUpload class to :
+
+   1) Validate the uploaded file (set the validation to true)
+   2) Run dicomTar.pl on the file (set the dicomTar to true)
+   3) Run tarchiveLoader on the file (set the minc-created to true)
+   4) Remove the uploaded file once the previous steps have completed
+   5) Update the mri_upload table
+
+Documentation: perldoc imaging_upload_file.pl
+
+HELP
+my $Usage = <<USAGE;
+usage: $0 </path/to/UploadedFile> -upload_id [options]
+       $0 -help to list options
+Note:  Please make sure that the </path/to/UploadedFile> and the upload_id
+provided correspond to the same upload entry.
+USAGE
+
+&Getopt::Tabular::SetHelp( $Help, $Usage );
+&Getopt::Tabular::GetOptions( \@opt_table, \@ARGV )
+    || exit $NeuroDB::ExitCodes::GETOPT_FAILURE;
+
+################################################################
+############### Establish database connection ##################
+################################################################
+{ package Settings; do "$ENV{LORIS_CONFIG}/.loris_mri/$profile" }
+my $dbh = &NeuroDB::DBI::connect_to_db(@Settings::db);
+
+################################################################
+################## Instantiate MRIProcessingUtility ############
+################################################################
+my $data_dir = NeuroDB::DBI::getConfigSetting(
+    \$dbh,'dataDirBasepath'
+);
+
+my $template         = "ProtocolCheck-XXXX"; # for tempdir
+my $TmpDir = tempdir(
+    $template, TMPDIR => 1, CLEANUP => 1
+);
+
+my @temp     = split(/\//, $TmpDir);
+my $templog  = $temp[$#temp];
+my $LogDir   = "$data_dir/logs";
+if (!-d $LogDir) {
+    mkdir($LogDir, 0770);
+}
```

Comment:
What if the directory exist but the permissions are not properly set?
Should also check if the script can write to that directory